source/FreeRTOS_DNS_Cache.c

```diff
@@ -367,5 +439,134 @@
+
+            #if ( ipconfigDNS_CACHE_ADDRESSES_PER_ENTRY > 1 )
+                uxNumIPAddresses = ( size_t ) xDNSCache[ xIndex ].ucNumIPAddresses;
+
+                if( uxNumIPAddresses > ipconfigDNS_CACHE_ADDRESSES_PER_ENTRY )
+                {
+                    /* Make this a configASSERT()? */
+                    uxNumIPAddresses = ipconfigDNS_CACHE_ADDRESSES_PER_ENTRY;
+                }
+            #endif /* ( ipconfigDNS_CACHE_ADDRESSES_PER_ENTRY > 1 ) */
+
+            for( uxIPAddressIndex = 0; uxIPAddressIndex < uxNumIPAddresses; uxIPAddressIndex++ )
+            {
+                pxAddresses = &( xDNSCache[ xIndex ].xAddresses[ uxIPAddressIndex ] );
+
+                #if ( ipconfigUSE_IPv6 != 0 )
+                    if( pxAddresses->xIs_IPv6 != pdFALSE )
+                    {
+                        pxNewAddress = pxNew_AddrInfo( xDNSCache[ xIndex ].pcName, FREERTOS_AF_INET6, pxAddresses->xAddress_IPv6.ucBytes );
+                    }
+                    else
+                #endif /* ( ipconfigUSE_IPv6 != 0 ) */
+                {
+                    uint8_t * ucBytes = ( uint8_t * ) &( pxAddresses->ulIPAddress );
+
+                    pxNewAddress = pxNew_AddrInfo( xDNSCache[ xIndex ].pcName, FREERTOS_AF_INET4, ucBytes );
+                }
+
+                if( pxNewAddress != NULL )
+                {
+                    if( *( ppxAddressInfo ) == NULL )
+                    {
+                        /* For the first address found. */
+                        *( ppxAddressInfo ) = pxNewAddress;
+                    }
+                    else
+                    {
+                        /* For the next address found. */
+                        *( ppxLastAddress ) = pxNewAddress;
+                    }
+
+                    ppxLastAddress = &( pxNewAddress->ai_next );
+                }
+            }
+        }
+    #endif /* #if( ipconfigUSE_DNS_CACHE == 1 ) */
+/*-----------------------------------------------------------*/
+
+    #if ( ipconfigUSE_DNS_CACHE == 1 )
+        uint32_t Prepare_CacheLookup( const char * pcHostName,
+                                      BaseType_t xFamily,
+                                      struct freertos_addrinfo ** ppxAddressInfo )
+        {
+            uint32_t ulIPAddress = 0U;
+
+            if( xFamily == FREERTOS_AF_INET6 )
+            {
+                IPv46_Address_t xIPv46_Address;
+                BaseType_t xFound;
+
+                xIPv46_Address.xIs_IPv6 = pdTRUE;
+                xFound = FreeRTOS_ProcessDNSCache( pcHostName, &( xIPv46_Address ), 0, pdTRUE, ppxAddressInfo );
+
+                if( xFound != 0 )
+                {
+                    if( ( ppxAddressInfo != NULL ) && ( *( ppxAddressInfo ) != NULL ) )
+                    {
+                        struct freertos_sockaddr6 * sockaddr6 = ( ( sockaddr6_t * ) ( *( ppxAddressInfo ) )->ai_addr );
+
+                        ( void ) sockaddr6;
```

Comment:
These two lines were used for inline-debugging. I think you can remove them:
~~~diff
-    struct freertos_sockaddr6 * sockaddr6 = ( ( sockaddr6_t * ) ( *( ppxAddressInfo ) )->ai_addr );
-
-    ( void ) sockaddr6;
~~~